docs/cep/reference/functions/adhoc/adhoc.md

```diff
@@ -0,0 +1,78 @@
+---
+title: Adhoc Query
+---
+
+Adhoc queries provide a way of performing ad-hoc operations on tables (stores), named-windows, and named-aggregations. We can send adhoc queries and fetch data from Stores and named windows.
+
+
+
+## Syntax
+
+Adhoc Query uses the following syntax:
+
+```js
+	select * from SampleAdhocQueryInputTableOneMinTimeWindow;
+	
+	SELECT * FROM SampleAdhocQueryTable;
+```				  
+					  
+
+
+## Example
+
+```js
+-- Defines `SampleAdhocQueryInputTable` collection to process events having `sensorId` and `temperature`(F).
+CREATE SOURCE SampleAdhocQueryInputTable WITH(type = 'database', collection = "SampleAdhocQueryInputTable", collection.type="doc" , replication.type="global", map.type='json') (sensorId string, temperature double);
+
+-- Named Window
+CREATE WINDOW SampleAdhocQueryInputTableOneMinTimeWindow (sensorId string, temperature double) SLIDING_TIME(1 min);
+
+-- Table
+CREATE TABLE SampleAdhocQuerySensorA1234DestTable(sensorId string, temperature double);
+
+@info(name = 'Insert-to-window')
+INSERT INTO SampleAdhocQueryInputTableOneMinTimeWindow
+SELECT *
+FROM SampleAdhocQueryInputTable;
+
+@info(name = 'EqualsFilter')
+-- Note: Filter out events with `sensorId` equalling `sensor A1234`
+INSERT INTO SampleAdhocQuerySensorA1234DestTable
+SELECT *
+FROM SampleAdhocQueryInputTable
+WHERE sensorId == 'sensor A1234';
+```
+
+
+1. Upload following data into `SampleAdhocQueryInputTable` C8DB Collection
```

Comment:
What are these? Do the numbers correspond to parts of the code block above? I am very confused.

Reply:
Also, where do I do any of this? Query editor? Stream worker screen?